### PR TITLE
feat: use webpack caching in development

### PR DESCRIPTION
**Pull request by ojeytonwilliams**
This cuts build time for build-workers considerably.

I also removed the echos.  In part because they made it hard to pass args around and in part because I don't think they're very useful.

Finally, I took out `build:workers:debug` because I've never used it.

### CODE REVIEW

File: client/webpack-workers.js [hunk 1]
```diff
@@ -8,6 +8,7 @@ module.exports = (env = {}) => {
   const staticPath = path.join(__dirname, './static/js');
   const configPath = path.join(__dirname, '../config/client');
   return {
+    cache: __DEV__ ? { type: 'filesystem' } : false,
```
Comment (ShaunSHamilton) [inline client/webpack-workers.js:11]: If we have no additional config for `cache`, why are we not using `type: 'memory'`? https://webpack.js.org/configuration/other-options/#cachetype

_You will probably need to explain like I am 5_

Comment (raisedadead) [reply client/webpack-workers.js:11]: To me, it looks like, that way the cache persists between hard restarts of the development cycles?

Comment (ojeytonwilliams) [reply client/webpack-workers.js:11]: From my understanding, it literally uses RAM.  So, yeah, it won't persist.

**Review by raisedadead — DISMISSED** [review]

**Review by ShaunSHamilton — COMMENTED** [review]
Shall I open a new PR for updating Gitpod to use this? Or, can it be done in this one?

**Comment by raisedadead** [comment]
Feel free to add a commit here, and merge.

**Comment by ShaunSHamilton** [comment]
**Just for radar:** Gatsby makes a potentially useful point here:
```console
Hi from the Gatsby maintainers! Based on what we see in your site, these coming
features may help you. All of these can be enabled within gatsby-config.js via
flags (samples below)

Preserve webpack's Cache (https://github.com/gatsbyjs/gatsby/discussions/28331), which
changes Gatsby's cache clearing behavior to not clear webpack's
cache unless you run "gatsby clean" or delete the .cache folder manually.
Here's how to try it:

module.exports = {
  flags: { PRESERVE_WEBPACK_CACHE: true },
  plugins: [...]
}
```
Obviously, not before it is ready, but if we are storing the webpack cache on the filesystem for persistence, then we may not want Gatsby to clear it.

**Comment by ojeytonwilliams** [comment]
@ShaunSHamilton I think we're okay: I tried `gatsby clean` and it didn't touch this webpack cache. AFAIK, all the Gatsby cache files, including webpack, are held separately in client/.cache

**Review by kzaw20081 — COMMENTED** [review]
Good

**Pull request merged**